src/sdk/service/iam-service.ts

```diff
@@ -151,4 +157,97 @@ export class IamService {
     }
     return false;
   }
+
+  /**
+   * Validate a role creation request.
+   * @param {RoleCreationRequest} roleCreationRequest
+   * @param {CompanyInventory} companyInventory
+   * @returns {Array<Error>}
+   */
+  static validateRole(roleCreationRequest: RoleCreationRequest, companyInventory: CompanyInventory) {
+    const errors: Array<Error> = [];
+    if (roleCreationRequest.policies.length === 0) {
+      errors.push((new Error('A role must have at least one policy.')));
+    }
+    for (const policy of roleCreationRequest.policies) {
+      const entityOptional: InventoryEntity | undefined = companyInventory.getEntityByUuid(policy.entityUuid);
+      if (entityOptional === undefined) {
+        errors.push((new Error('Entity not found in this company.')));
```

Comment:
need to include the name of the entity in these so they're useful to the user

Reply:
We can't get the entity name but only the Entity UUID there. Because the entity doesn't exist so we can't get the name ;-). But i'll put the UUID then.